### ISSUE TITLE
Fatal error on saving lessons (with content drip)

### ISSUE THREAD

**Issue by wdfee** [woocommerce/sensei-content-drip#124]
Since update to version 1.0.8 it is not possible to edit lessons anymore, where dripping has been added before. Error:

> Catchable fatal error: Object of class DateTime could not be converted to string in /[...]/wp-includes/formatting.php on line 1031

(could have been 1.0.7, too, we used 1.0.6 before updating)

**Comment by wdfee** [comment]
found it: class-scd-ext-lesson-admin.php line 494:
esc_html() can only be used on strings, so the condition needs some finetuning:
```
if ( ! empty( $value ) && is_string( $value ) ) {
		$lesson_drip_data[ $field_key ] = esc_html( $value );
}
```
this removes the fatal error on saving

**Comment by donnapep** [comment]
This was introduced in 1.0.7 by https://github.com/woocommerce/sensei-content-drip/pull/106. While I can't reproduce the fatal, I agree we should do some type-checking. [This code](https://github.com/woocommerce/sensei-content-drip/issues/124#issuecomment-291840926) would not assign a value to `$lesson_drip_data[ $field_key ]` for a non-string value though, so is not a complete solution.

**Comment by alexsanford** [comment]
Closed by #148
